pkg/controller/service.go

```diff
@@ -87,7 +89,12 @@ func (s *ShadowServiceManager) CreateOrUpdate(svc *corev1.Service) (*corev1.Serv
 	}
 
 	// If the kubernetes server version is 1.17+, then use the topology key.
-	if major, minor := parseKubernetesServerVersion(s.kubeClient); major == 1 && minor >= 17 {
+	serverVersion, err := getServerVersion(s.kubeClient)
```

Comment:
What do you think about doing it only once in the `NewShadowServiceManager`? `kubeClient.Discovery().ServerVersion()` is making a request on the server each time we call it.

Reply:
Why not 🙂

Reply:
After trying to do it, it appears that we have to handle the error through all the construction chain, which is not ideal. My thinking is that it's better to keep it like that for now.